docs/dev/oauth_for_data_integrations.md

```diff
@@ -0,0 +1,116 @@
+
+7. To validate that OAuth is working properly, make a request to the `marketplace_create_api` endpoint with your token in the headers of the request. 
+
+   If successful, this request returns an API key that you should securely save so you can use it to submit data into Datadog on behalf of the user. **You cannot access this API key value again after the initial request response**.
+
+8. Test any additional scopes that you’ve requested access for.
+
+### Publish the OAuth client
+
+In order to publish an OAuth client, you need to open a pull request for your data integration in either the [`integrations-extras`][5] or [Marketplace][6] GitHub repositories. 
+
+To start the publishing process in the [Developer Platform][4]:
+
+1. Navigate to the **Publishing** tab under **General** to receive your published client ID, client secret, and `app_uuid` to use for publishing. Enter additional information about your integration in the fields.
+
+   Save your client ID, client secret, and `app_uuid` in a secure location. 
+
+2. When opening a pull request in `integrations-extras` or `Marketplace`, use the `app_uuid` value for publishing in the `app_uuid` field of the `manifest.json` file. If the `app_uuid` values do not align, your application does not publish correctly.
+
+You cannot edit a published OAuth client directly, so only go through the publishing flow when everything has been tested and is ready to go. To make updates to the OAuth client, you need to go through the publishing flow again. **The published client credentials do not appear again**.
+
+For more information about the integration publishing process, see the [Marketplace and Integrations documentation][7]. 
+
+## Add OAuth to an existing offering
+
+The process for adding an OAuth client to an existing integration is similar to what is outlined above, with some key differences.
+
+### If you have an existing data integration that’s not connected to a UI Extension
+
+Follow the [steps](#build-a-data-integration-with-oauth) above. Instead of opening a pull request, add a link to your existing file directory in the `integrations-extras` or Marketplace repository during the publishing process. 
+
+### If you have an existing data integration that’s currently connected to a UI Extension (shares the same tile)
+
+Instead of creating an app, navigate to an app that includes your published UI Extension in the Developer Platform and follow the remaining [steps](#create-an-oauth-client). You need to open a pull request to update the `app_uuid` field in the `manifest.json` file. 
+
+Once you’ve created your data integration's OAuth client and are ready for publishing, click **Edit** on your app and navigate to the **Publishing** tab under **General**. After completing the fields, you receive an `app_uuid` to use for publishing. Set this value as the `app_uuid` in your data integration's `manifest.json` file. 
```

Comment:
The answer is no -- we cannot update an existing app's uuid, so this will have to change. Only partners creating a new integration should update their PR's `app_uuid` field to match the `published_app_uuid` from the dev platform -- otherwise, we must do the reverse and change the private app's `published_app_uuid` to match the existing published app.

Reply:
Would there be any action that the partner has to take in that case where they have an existing app?

Reply:
@bgoldberg122  updated in latest commit - take a look and let me know if everything is accurate